sycl/include/sycl/ext/intel/esimd/detail/operators.hpp

```diff
@@ -482,6 +482,334 @@ __ESIMD_DEF_SIMD_VIEW_CMP_OP(>=, __ESIMD_DNS::is_simd_type_v<SimdT1>)
+  }                                                                            \
+  /* simd_mask CMPOP SCALAR */                                                 \
+  template <class SimdT1, class T1 = typename SimdT1::element_type, class T2,  \
+            class SimdT2 = typename __ESIMD_NS::simd<T2, SimdT1::length>,      \
+            class Tx = __ESIMD_DNS::computation_type_t<T1, T2>,                \
+            class SimdTx = __ESIMD_NS::simd<Tx, SimdT1::length>>               \
+  inline std::enable_if_t<__ESIMD_DNS::is_simd_mask_type_v<SimdT1> && COND,    \
+                          __ESIMD_NS::simd_mask<SimdT1::length>>               \
+  operator CMPOP(const SimdT1 &LHS, T2 RHS) {                                  \
+    return SimdTx(LHS) CMPOP RHS;                                              \
+  }                                                                            \
+                                                                               \
+  /* SCALAR CMPOP simd_obj_impl */                                             \
+  template <class T1, class SimdT2, class T2 = typename SimdT2::element_type,  \
+            class SimdT1 = typename __ESIMD_NS::simd<T1, SimdT2::length>,      \
+            class Tx = __ESIMD_DNS::computation_type_t<T1, T2>,                \
+            class SimdTx = __ESIMD_NS::simd<Tx, SimdT2::length>>               \
+  inline std::enable_if_t<__ESIMD_DNS::is_simd_mask_type_v<SimdT2> && COND,    \
+                          __ESIMD_NS::simd_mask<SimdT2::length>>               \
+  operator CMPOP(T1 LHS, const SimdT2 &RHS) {                                  \
+    return LHS CMPOP SimdTx(RHS);                                              \
+  }                                                                            \
+  /* simd_mask CMPOP simd_mask  */                                             \
+  template <class SimdT, class T1 = typename SimdT::element_type,              \
+            class T2 = typename SimdT::element_type>                           \
+  inline std::enable_if_t<__ESIMD_DNS::is_simd_mask_type_v<SimdT> && COND,     \
+                          __ESIMD_NS::simd_mask<SimdT::length>>                \
+  operator CMPOP(const SimdT &LHS, const SimdT &RHS) {                         \
+    return __ESIMD_NS::simd<T1, SimdT::length>(LHS)                            \
+        CMPOP __ESIMD_NS::simd<T2, SimdT::length>(RHS);                        \
+  }
+
+#define __ESIMD_CMP_OP_FILTER                                                  \
+  __ESIMD_DNS::is_valid_simd_elem_type_v<T1>                                   \
+      &&__ESIMD_DNS::is_valid_simd_elem_type_v<T2>
+// Equality comparison is defined for all simd_obj_impl subclasses.
+__ESIMD_DEF_SIMD_MASK_CMP_OP(==, __ESIMD_CMP_OP_FILTER)
+__ESIMD_DEF_SIMD_MASK_CMP_OP(!=, __ESIMD_CMP_OP_FILTER)
+
+// Relational operators are defined only for the simd type.
```

Comment:
This comment confused me a bit. 
What type combination is supported for '==' at L673 above and not supported for '<' on L677 below?